### PR TITLE
ENT-6914 Upgraded to Kotlin 1.8/JDK 17/Gradle 7.6

### PR DESCRIPTION
**Pull request by arshadm**
This PR updates this project for Corda 4.12.

The major changes are:
1. Gradle 7.6
2. Kotlin 1.8
3. JDK17
4. Remove circular dependency on token-sdk, this has meant the integration tests in this project have moved to the token-sdk

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -60,22 +85,62 @@ allprojects {
...
+            }
+        }
     }
 
     tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile) {
         kotlinOptions {
-            languageVersion = "1.2"
-            apiVersion = "1.2"
-            jvmTarget = "1.8"
+            languageVersion = "1.8"
```
Comment (adelel1) [inline build.gradle:141]: Should be 1.9 now?

File: workflows/src/integrationTest/kotlin/com/r3/corda/lib/ci/tests/DriverBasedTest.kt
Comment (adelel1) [inline workflows/src/integrationTest/kotlin/com/r3/corda/lib/ci/tests/DriverBasedTest.kt:1]: Why were these moved?

File: build.gradle (hunk 1 above)
Comment (arshadm) [reply build.gradle:141]: Renamed

File: workflows/src/integrationTest/kotlin/com/r3/corda/lib/ci/tests/DriverBasedTest.kt
Comment (arshadm) [reply workflows/src/integrationTest/kotlin/com/r3/corda/lib/ci/tests/DriverBasedTest.kt:1]: Remove circular dependency on token-sdk

**Review by ronanbrowne — APPROVED** [review]
LGTM - ( jenkins changes)

**Review by adelel1 — APPROVED** [review]
